lib/library.rb

```diff
@@ -0,0 +1,41 @@
+class Library
+  attr_reader :name, :books, :authors, :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+  end
+
+  def add_author(author)
+    @authors << author
+    author.books.each do |book|
+      @books << book
+    end
+  end
+
+  def publication_time_frame_for(author)
+    books = author.books.minmax_by do |book|
```

Comment:
Logic here is great, but this method could be split up to stick to SRP better.  What might it look like if an author could give you a list of their books sorted by publication year?